### PR TITLE
fix(_get_all_snapshot_files): modified to suit azure

### PR DESCRIPTION
**Pull request by ShlomiBalalis**
Previously, the function could only list files for aws and gce. Because of that,
the test that uses this function, test_backup_purge_removes_orphan_files, constatnly
failed when the job used azure as a backend for the backup.
I added a new function, _get_all_snapshot_files_azure, to remedy that.

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by fgelcer — COMMENTED** [review]
only need to fix the `docker/env/version` once you are done with your tests, other than that, LGTM

**Comment by ShlomiBalalis** [comment]
> only need to fix the `docker/env/version` once you are done with your tests, other than that, LGTM

@fgelcer wdym? That's what @bentsi told me I can name it. Should I give it another name?

I'm done with the tests btw, all passed

**Review by fgelcer — APPROVED** [review]
LGTM

File: mgmt_cli_test.py [hunk 1]
```diff
@@ -552,16 +553,30 @@ def _get_all_snapshot_files_gce(cluster_id, bucket_name):
...
+    def _get_all_snapshot_files_azure(cluster_id, bucket_name):
+        file_set = set()
+        credentials = KeyStore().get_backup_azure_blob_credentials()
+        blob_service_client = BlobServiceClient(account_url=f"https://{credentials['account']}.blob.core.windows.net",
+                                                credential=credentials['key'])
+        container_client = blob_service_client.get_container_client(bucket_name)
+        blob_iterator = container_client.list_blobs(name_starts_with=f"backup/sst/cluster/{cluster_id}")
+        for listing_object in blob_iterator:
+            file_set.add(listing_object["name"])
+        return file_set
```
Comment (enaydanov) [inline mgmt_cli_test.py:566]: Have you thought about using libcloud for the same task?

```python
credentials = KeyStore().get_backup_azure_blob_credentials()
azure_driver = libcloud.storage.providers.get_driver(libcloud.storage.types.Provider.AZURE_BLOBS)
driver = azure_driver(key=credentials["account"], secret=credentials["key"])
```
and rest is the same as for GCE.

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: mgmt_cli_test.py (hunk 1 above)
Comment (ShlomiBalalis) [reply mgmt_cli_test.py:566]: You're right, it's working finely!
Updating

**Review by enaydanov — APPROVED** [review]

**Review by roydahan — APPROVED** [review]

**Pull request merged**